integration/src/main/java/scripts/utils/GcsBucketAccessTester.java

```diff
@@ -104,45 +165,95 @@ public void checkAccess(TestUserSpecification testUser, @Nullable ControlledReso
-      assertFalse(doWithOptionalWait(this::blobUpdate), "no role cannot update");
-      assertFalse(doWithOptionalWait(this::blobDelete), "no role cannot delete");
-      assertFalse(doWithOptionalWait(this::bucketGet), "no role cannot bucket get");
-      assertFalse(doWithOptionalWait(this::bucketDelete), "no role cannot bucket delete");
-      return;
+      if (doNoWait(() -> blobCreate(testClient))) {
+        return "no role cannot create";
+      }
+      if (doNoWait(this::blobRead)) {
+        return "no role cannot read";
+      }
+      if (doNoWait(this::blobUpdate)) {
+        return "no role cannot update";
+      }
+      if (doNoWait(this::blobDelete)) {
+        return "no role cannot delete";
+      }
+      if (doNoWait(this::bucketGet)) {
+        return "no role cannot bucket get";
+      }
+      if (doNoWait(this::bucketDelete)) {
+        return "no role cannot bucket delete";
+      }
+      return null; // Success
     }
 
     switch (role) {
-      case READER:
-        assertTrue(doWithOptionalWait(this::blobRead), "reader can read");
-        assertFalse(doWithOptionalWait(() -> blobCreate(testClient)), "reader cannot create");
-        assertFalse(doWithOptionalWait(this::blobUpdate), "reader cannot update");
-        assertFalse(doWithOptionalWait(this::blobDelete), "reader cannot delete");
-        assertFalse(doWithOptionalWait(this::bucketGet), "reader cannot bucket get");
-        assertFalse(doWithOptionalWait(this::bucketDelete), "reader cannot bucket delete");
-        return;
-
-      case WRITER:
-        assertTrue(doWithOptionalWait(() -> blobCreate(testClient)), "writer can create");
-        assertTrue(doWithOptionalWait(this::blobRead), "writer can read");
-        assertTrue(doWithOptionalWait(this::blobUpdate), "writer can update");
-        assertTrue(doWithOptionalWait(this::blobDelete), "writer can delete");
-        assertFalse(doWithOptionalWait(this::bucketGet), "writer cannot bucket get");
-        assertFalse(doWithOptionalWait(this::bucketDelete), "writer cannot bucket delete");
-        return;
-
-      case EDITOR:
-        assertTrue(doWithOptionalWait(() -> blobCreate(testClient)), "editor can create");
-        assertTrue(doWithOptionalWait(this::blobRead), "editor can read");
-        assertTrue(doWithOptionalWait(this::blobUpdate), "editor can update");
-        assertTrue(doWithOptionalWait(this::blobDelete), "editor can delete");
-        assertTrue(doWithOptionalWait(this::bucketGet), "editor can bucket get");
-        assertFalse(doWithOptionalWait(this::bucketDelete), "editor cannot bucket delete");
-        return;
-
-      default:
-        throw new IllegalArgumentException("Invalid IAM role: " + role);
+      case READER -> {
+        if (!doWithOptionalWait(this::blobRead)) {
+          return "reader can read";
+        }
+        if (doNoWait(() -> blobCreate(testClient))) {
+          return "reader cannot create";
```

Comment:
why does it return "cannot create", when doNoWait return true, I thought that means the action succeeded?